Tests/BackendIntegrationTests/BaseStoreKitIntegrationTests.swift

```diff
@@ -147,19 +147,27 @@ extension BaseStoreKitIntegrationTests {
         file: FileString = #file,
         line: UInt = #line
     ) async throws -> EntitlementInfo {
-        let entitlements = customerInfo.entitlements.all
+        // This is used to throw an error when the test fails.
+        // For some reason XCTest is continuing execution even after a test failure
+        // despite having `self.continueAfterFailure = false`
+        //
+        // By doing this, instead of only calling `fail`, we ensure that
+        // Swift stops executing code when an assertion has failed,
+        // and therefore avoid code running after the test has already failed.
+        // This prevents test crashes from code calling `Purchases.shared` after the test has ended.
+        func failTest(_ message: String) async throws {
```

Comment:
Is this happening in other tests?

Reply:
Not that I know. This particular case was happening a fair amount in CI though, because we use this method in pretty much all integration tests.